### PR TITLE
Feat : Added Update profile

### PR DESCRIPTION
**Pull request by srejitk**
The following merge request adds the following features:

1. Update Profile Page
2. Edit Post Modal
3. Profile Page
4. User Profile Page

### CODE REVIEW

File: social-circus/src/redux/actions/uploadImageActions.jsx [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
+export const UploadImage = async (path, file) => {
+  try {
+    const imageRef = ref(storage, path);
+    const response = await uploadBytesResumable(imageRef, file);
+    const pathName = response?.ref?.toString();
+    const gsReference = ref(storage, pathName);
+    const url = await getDownloadURL(gsReference);
+    return url;
+  } catch (error) {
+    console.log(error);
```
Comment (vaibhav-mahalle) [inline social-circus/src/redux/actions/uploadImageActions.jsx:14]: you can add ToastMsg here.

**Review by vaibhav-mahalle — APPROVED** [review]
Loved the structured commits of your pr. Code is clean, suggested a minor change. Please look into it.

File: social-circus/src/components/EditPostModal/EditPostModal.jsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    }
+    setForm(initialValues);
+    setOpenEmoji(false);
+    dispatch(getExplorePosts());
+  };
+
+  const handleImage = async (image) => {
+    if (image === null) return;
+    UploadImage(`posts/${token}`, image);
+    //Please ignore the below code || WIP
```
Comment (shubhamkumar648) [inline social-circus/src/components/EditPostModal/EditPostModal.jsx:77]: remove commented code

**Review by shubhamkumar648 — APPROVED** [review]
- good code structure 
- perfect used of redux 
- suggestion : put some image or link for better description for pr ,it helps easy for review 
- good to go
-

File: social-circus/src/redux/actions/uploadImageActions.jsx (hunk 1 above)
Comment (srejitk) [reply social-circus/src/redux/actions/uploadImageActions.jsx:14]: Ah. Yes, cause the user wouldn't know if the image got uploaded. Nice Vaibhav.

File: social-circus/src/components/EditPostModal/EditPostModal.jsx (hunk 2 above)
Comment (srejitk) [reply social-circus/src/components/EditPostModal/EditPostModal.jsx:77]: Yes, Sir. Will do

**Pull request merged**